test/integration/model/schema.test.js

```diff
@@ -0,0 +1,458 @@
+'use strict';
+
+/* jshint -W030 */
+/* jshint -W110 */
+var chai = require('chai')
+  , expect = chai.expect
+  , Support = require(__dirname + '/../support')
+  , DataTypes = require(__dirname + '/../../../lib/data-types')
+  , current = Support.sequelize;
+var SCHEMA_ONE = 'schema_one';
+var SCHEMA_TWO = 'schema_two';
+
+var locationId;
+
+describe(Support.getTestDialectTeaser('Model'), function() {
+  if (current.dialect.supports.schemas) {
+
+    describe('schemas', function() {
+      before(function() {
+        this.Restaurant = current.define('restaurant', {
+            foo: DataTypes.STRING,
+            bar: DataTypes.STRING
+          },
+          {tableName: "restaurants"});
+        this.Location = current.define('location', {
+            name: DataTypes.STRING
+          },
+          {tableName: "locations"});
+        this.Employee = current.define('employee', {
+            first_name: DataTypes.STRING,
+            last_name: DataTypes.STRING
+          },
+          {tableName: "employees"});
+        this.EmployeeOne = this.Employee.schema(SCHEMA_ONE);
+        this.EmployeeTwo = this.Employee.schema(SCHEMA_TWO);
+        this.Restaurant.belongsTo(this.Location,
+          {
+            foreignKey: 'location_id',
+            constraints: false
+          });
+        this.Employee.belongsTo(this.Restaurant,
+          {
+            foreignKey: 'restaurant_id',
+            constraints: false
+          });
+        this.Restaurant.hasMany(this.Employee, {
+          foreignKey: 'restaurant_id',
+          constraints: false
+        });
+        this.RestaurantOne = this.Restaurant.schema(SCHEMA_ONE);
+        this.RestaurantTwo = this.Restaurant.schema(SCHEMA_TWO);
+      });
+
+
+      beforeEach('build restaurant tables', function() {
+        var self = this;
+        return current.createSchema('schema_one').then(function() {
+          return current.createSchema('schema_two');
+        }).then(function() {
+          return self.RestaurantOne.sync({force: true});
```

Comment:
return self.sequelize.sync()

And you don't need to `catch` - the test will fail if the chain is rejected